sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/BinaryDataHelper.java

```diff
@@ -50,8 +50,14 @@ public static void setAccessor(final BinaryDataAccessor binaryDataAccessor) {
      * @throws NullPointerException If {@code content} is null.
      */
     public static BinaryData createBinaryData(BinaryDataContent content) {
-        ensureAccessorSet();
-        return accessor.createBinaryData(content);
+        try {
+            ensureAccessorSet();
+            return accessor.createBinaryData(content);
+        } catch (Exception e) {
```

Comment:
May want to look into catching `Throwable` as we could get `Error`s causing this to fail